enhancements/installer/infrastructure-apis-custom-trust.md

```diff
@@ -0,0 +1,236 @@
+---
+title: infrastructure-apis-custom-trust
+authors:
+  - "@abhinavdahiya"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-06-30
+last-updated: 2020-06-30
+status: implementable
+see-also:
+  - "/enhancements/proxy/global-cluster-egress-proxy.md"  
+---
+
+# Custom Trust for Infrastructure APIs
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The users should be allowed to provide a custom trust bundle for the purpose of communicating with infrastructure APIs, which should be available to all the operators so that they can configure their SDKs. The operators will be able to pull the trust bundle from a defined ConfigMap or use the [`config.openshift.io/inject-trusted-cabundle`][egress-proxy-proposal] mechanism to mount a volume with trust already configured. The later is important because a majority of our operators communicate with infrastructure APIs and configuring the SDKs with custom trust is tedious and error prone similar to the PROXY setup.
+
+
+## Motivation
+
+OpenShift components interact quite a bit with the underlying infrastructure using APIs to provide it's users various features and a majority of our operators today interact with infrastructure APIs. For our commercial platforms like AWS, Azure, GCP's public cloud trusting the secured API endpoints is not a problem, but for other platforms like vSphere, OpenStack or even private clouds from AWS or Azure we need our operators to trust user-provided bundles to securely communicate to the APIs.
+
+### Goals
+
+1. Allow users to provide custom trust for APIs using openshift-installer.
+2. Allow users to provide custom trust for APIs as a day-2 operation.
+3. Provide a no hassle workflow for operators to consume the trust bundle on-par with [proxy][egress-proxy-proposal]
+
+### Non-Goals
+
+None at this time.
+
+## Proposal
+
+1. openshift-installer allows the users to provide custom trust bundle for infrastructure APIs using the `install-config.yaml`.
+2. [Infrastructure][openshift-api-infrastructure] object allows user to provide custom trust in the form of a ConfigMap in `openshift-config` namespace.
```

Comment:
> This assumes the nodes do not need to talk to the APIs before the cluster is up, however this may not be the case since kubelet likely needs to fetch information from the infrastructure to register the nodes.

I do not think this is making that assumption.
https://github.com/openshift/enhancements/pull/395/files#diff-c95f1e6520bccf9d5a26bb51ec5759a0R137
shows how MCO will use it to configure the hosts, so that covers the trust handling for host services.

And as for bootstrapping https://github.com/openshift/enhancements/pull/395/files#diff-c95f1e6520bccf9d5a26bb51ec5759a0R137 also provides detaild on how MCO will get the same information on bootstrap host.